test/quantization/fx/test_quantize_fx.py

```diff
@@ -2543,6 +2547,232 @@ def forward(self, x):
             mp(torch.rand(4, 4, 4, 4))
             mc = convert_fx(mp)
 
+
+
+    class _NonReferenceTestModel(nn.Module):
+        def __init__(self, func, lin_in, lin_out):
+            super().__init__()
+            self.conv1 = nn.Conv2d(3, 6, 5)
+            self.pool = nn.MaxPool2d(2, 2)
+            self.lin = nn.Linear(lin_in, lin_out)
+            self.func = func
+
+        def forward(self, x, y, z):
+            x = self.pool(F.relu(self.conv1(x)))
+            x = torch.flatten(x, 1)
+            x = self.func(x, y, z)
+            x = self.lin(x)
+            return x
+
+    # This function looks at the node specified by the NodeInfo in the key of
+    # node_info_to_non_tensor_args and checks that the args at specified indices
+    # are not observed (since they are non tensors). If the args at those indices
+    # are a tuple/list (which do not show up as nodes) the function checks the
+    # individual elements of the tuple/list recursively.
+    def _check_not_observed(self, model, node_info_to_non_tensor_args):
+
+        # this is a helper function (for easier recursion) that looks at specific
+        # indices of nodes and checks whether they are observed
+        def _check_node_indices_not_observed(model, node, indices):
+            for index in indices:
+                if hasattr(node, "args"):
```

Comment:
this looks a bit weird, is it possible to remove this check?

Reply:
no, otherwise it fails with tuple arg types